lightstep-tracer-jre/src/benchmark/java/com/lightstep/benchmark/BenchmarkClient.java

```diff
@@ -0,0 +1,237 @@
+	    throw new RuntimeException(e);
+	} catch(IOException e) {
+	    throw new RuntimeException(e);
+	}
+    }
+
+    <T> T postGetJson(String path, Class<T> cl) {
+	try (BufferedReader br = new BufferedReader(new InputStreamReader(getUrlReader(path)))) {
+	    return objectMapper.readValue(br, cl);
+	} catch(IOException e) {
+	    throw new RuntimeException(e);
+	}
+    }
+
+    Control getControl() {
+	return postGetJson("/control", Control.class);
+    }
+
+    void postResult(Result r) {
+	StringBuilder rurl = new StringBuilder("/result?timing=");
+	rurl.append(r.runTime);
+	rurl.append("&flush=");
+	rurl.append(r.flushTime);
+	rurl.append("&a=");
+	rurl.append(r.answer);
+	rurl.append("&s=");
+	for (Long sleep : r.sleepNanos) {
+	    rurl.append(sleep.toString());
+	    rurl.append(",");
+	}
+	try (BufferedReader br = new BufferedReader(new InputStreamReader(getUrlReader(rurl.toString())))) {
+	} catch(IOException e) {
+	    throw new RuntimeException(e);
+	}
+    }
+
+    static final long primeWork = 982451653;
+
+    long work(long n) {
+	long x = primeWork;
+	for (; n != 0; --n) {
+	    x *= primeWork;
+	}
+ 	return x;
+    }
+
+    OneThreadResult testBody(Control c, Tracer t) {
+	OneThreadResult r = new OneThreadResult();
+	r.sleepNanos = new ArrayList<Long>();
+
+	long sleepDebt = 0;
+
+ 	for (long i = 0; i < c.Repeat; i++) {
+	    Span span = t.buildSpan("span/test").start();
+	    r.answer = work(c.Work);
+
+	    for (long l = 0; l < c.NumLogs; l++) {
+		span.log("testlog", logPayloadStr.substring(0, (int)c.BytesPerLog));
+	    }
+
+	    span.finish();
+	    sleepDebt += c.Sleep;
+
+	    if (sleepDebt <= c.SleepInterval) {
+		continue;
+	    }
+
+	    long beginSleep = System.nanoTime();
+	    try {
+		long millis = sleepDebt / 1000000;
+		Thread.sleep(millis);
+	    } catch (InterruptedException e) {
+	    }
+
+	    long endSleep = System.nanoTime();
+	    long slept = endSleep - beginSleep;
+	    sleepDebt -= slept;
+	    r.sleepNanos.add(slept);
+ 	}
+
+	return r;
+    }
+
+    Result runTest(Control c) {
+	Tracer tracer;
+	if (c.Trace) {
+	    tracer = testTracer;
+	} else {
+	    tracer = new NoopTracer();
+	}
+
+	System.gc();
+
+	Result res = new Result();
+	int conc = c.Concurrent;
+
+	ArrayList<OneThreadResult> results = new ArrayList<OneThreadResult>();
+	long beginTest = System.nanoTime();
+
+	if (conc == 1) {
+	    results.add(testBody(c, tracer));
+	} else {
+	    ArrayList<Thread> threads = new ArrayList<Thread>();
+	    for (int i = 0; i < conc; i++) {
+		Thread th = new Thread() {
+			public void run() {
+			    OneThreadResult tr = testBody(c, tracer);
+			    synchronized (results) {
+				results.add(tr);
+			    }
+			}};
+		th.start();
+		threads.add(th);
+	    }
+	    for (Thread th : threads) {
+		try {
+		    th.join();
+		} catch (InterruptedException e) {
+		    throw new RuntimeException(e);
+		}
+	    }
+	}
+ 	long endTest = System.nanoTime();
+ 	if (c.Trace) {
+	    ((JRETracer)tracer).flush();
```

Comment:
You mean the real tracer that is idle during non-traced tests? Yes, I suppose. The same issue is present in several of the other benchmarks, and it shouldn't greatly impact the numbers.